hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/DynamicConfigTemplate.java

```diff
@@ -0,0 +1,246 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.template;
+
+import com.hazelcast.annotation.GenerateCodec;
+import com.hazelcast.annotation.Nullable;
+import com.hazelcast.annotation.Request;
+import com.hazelcast.annotation.Since;
+import com.hazelcast.client.impl.protocol.constants.ResponseMessageConst;
+import com.hazelcast.client.impl.protocol.task.dynamicconfig.EvictionConfigHolder;
+import com.hazelcast.client.impl.protocol.task.dynamicconfig.ListenerConfigHolder;
+import com.hazelcast.client.impl.protocol.task.dynamicconfig.MapStoreConfigHolder;
+import com.hazelcast.client.impl.protocol.task.dynamicconfig.NearCacheConfigHolder;
+import com.hazelcast.client.impl.protocol.task.dynamicconfig.QueryCacheConfigHolder;
+import com.hazelcast.client.impl.protocol.task.dynamicconfig.QueueStoreConfigHolder;
+import com.hazelcast.client.impl.protocol.task.dynamicconfig.RingbufferStoreConfigHolder;
+import com.hazelcast.config.CacheSimpleConfig.ExpiryPolicyFactoryConfig.TimedExpiryPolicyFactoryConfig;
+import com.hazelcast.config.CacheSimpleEntryListenerConfig;
+import com.hazelcast.config.HotRestartConfig;
+import com.hazelcast.config.MapAttributeConfig;
+import com.hazelcast.config.MapIndexConfig;
+import com.hazelcast.config.WanReplicationRef;
+import com.hazelcast.nio.serialization.Data;
+
+import java.util.List;
+
+@GenerateCodec(id = TemplateConstants.DYNAMIC_CONFIG_TEMPLATE_ID, name = "DynamicConfig", ns = "Hazelcast.Client.Protocol.Codec")
+@Since("1.5")
+public interface DynamicConfigTemplate {
+
+    /**
+     * Adds a new multimap config to a running cluster.
+     * If a multimap configuration with the given {@code name} already exists, then
+     * the new multimap config is ignored and the existing one is preserved.
+     *
+     * @param name              multimap configuration name
+     * @param collectionType    value collection type. Valid values are "SET" and "LIST".
+     * @param listenerConfigs   entry listener configurations
+     * @param binary            {@code true} to store values in {@code BINARY} format or {@code false} to store
+     *                          values in {@code OBJECT} format.
+     * @param backupCount       number of synchronous backups
+     * @param asyncBackupCount  number of asynchronous backups
+     * @param statisticsEnabled set to {@code true} to enable statistics on this multimap configuration
+     */
+    @Since("1.5")
```

Comment:
do not need this extra since tag because the interface already has it and the method should inherit it from interface when generating the documentation.